### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by margarita-ignatieva**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -17,6 +12,7 @@ public Rectangle(Integer width, Integer length, String color) {
     }
 
     public Integer getWidth() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +23,41 @@ public Integer getLength() {
...
+            return false;
+        }
+
+        if (obj.getClass() != this.getClass()) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) obj;
+        if ((width == rectangle.width) && (length != rectangle.length)
+                && (color.equals(rectangle.getColor()))) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: so it can return true even if length doesn't equal?

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Rectangle.java]: what if color would be null?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +23,41 @@ public Integer getLength() {
...
+            return true;
+        }
+
+        if ((width == null && rectangle.width == null)
+                && (length == null && rectangle.length == null)
+                && (color == null && rectangle.color == null)) {
+            return true;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: can we simplify that?

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (margarita-ignatieva) [reply src/main/java/core/basesyntax/Rectangle.java]: done

File: src/main/java/core/basesyntax/Rectangle.java (hunk 3 above)
Comment (margarita-ignatieva) [reply src/main/java/core/basesyntax/Rectangle.java]: done

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (margarita-ignatieva) [reply src/main/java/core/basesyntax/Rectangle.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**